--- origsrc/gap4r7/Makefile.in	2014-05-24 16:02:53.000000000 -0400
+++ src/gap4r7/Makefile.in	2014-07-19 18:10:48.330703300 -0400
@@ -130,7 +130,7 @@ extern: config
 
 compile: extern
 	( cd bin/$(GAPARCH); \
-		$(MAKE) \
+		$(MAKE) gapdll \
 		ABI='$(ABI)' \
 		CC='$(CC)' \
 		GMP_VER='$(GMP_VER)' \
--- origsrc/gap4r7/cnf/Makefile	2014-07-19 17:37:38.980919000 -0400
+++ src/gap4r7/cnf/Makefile	2014-07-19 18:11:32.332220100 -0400
@@ -111,7 +111,7 @@ Makegap.in: Makegap.top Makefile Makegap
 	@echo 'gapdll: $$(OBJECTS) $$(ITANIUMOBJ) $$(EXTOBJS) $$(GMP_LIBS)' >> Makegap.in
 	@echo '	$$(CC) $$(CPPFLAGS) $$(GMP_CFLAGS) $$(CFLAGS) -DCOMPILECYGWINDLL -o gap.o -c ../../src/gap.c' >> Makegap.in
 	@echo '	$$(CC) $$(LDFLAGS) -o gap.dll -shared $$(OBJECTS) $$(ITANIUMOBJ) -lm $$(MPILIBS) $$(EXTOBJS) $$(GMP_LIBS) $$(CONFLIBS)' >> Makegap.in
-	@echo '	$$(CC) $$(LDFLAGS) -o gapw95 ../../src/gapw95.c $$(CPPFLAGS) $$(GMP_CFLAGS) $$(CFLAGS) -DCOMPILECYGWINDLL gap.dll' >> Makegap.in
+	@echo '	$$(CC) $$(LDFLAGS) -o gap ../../src/gapw95.c $$(CPPFLAGS) $$(GMP_CFLAGS) $$(CFLAGS) -DCOMPILECYGWINDLL gap.dll' >> Makegap.in
 	@echo >> Makegap.in
 	@echo '# dependencies are automatically generated' >> Makegap.in
 	@(for i in `eval 'echo $(SOURCE)'`; do \
