--- origsrc/gap4r6/cnf/aclocal.m4	2013-07-20 16:02:01.000000000 -0400
+++ src/gap4r6/cnf/aclocal.m4	2013-08-22 17:02:18.119390900 -0400
@@ -148,7 +148,8 @@ AC_DEFUN(GP_CFLAGS,
         gp_cv_cflags="-O";;
    esac 
  ])
-CFLAGS="$CFLAGS $gp_cv_cflags"])
+# Put user-supplied CFLAGS last, don't override them
+CFLAGS="$gp_cv_cflags $CFLAGS"])
 
 dnl #########################################################################
 dnl ##
