--- pkg/InstPackages.sh.orig	2013-03-20 12:18:58.000000000 -0400
+++ pkg/InstPackages.sh	2013-08-08 08:52:55.688924300 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh -x 
+#!/bin/sh -x
 
 # You need 'gzip', GNU 'tar', a C compiler, sed, pdftex to run this.
 # Copy this script to the 'pkg' subdirectory of your GAP installation,
@@ -32,7 +32,7 @@
 
 cd Browse
 ./configure
-make
+make CFLAGS="-I/usr/include/ncursesw -L/usr/lib/ncursesw"
 cd ..
 
 # TODO: FIX Carat
