--- origsrc/gap4r7/InstPackages.sh	2014-07-19 17:37:38.435887800 -0400
+++ src/gap4r7/InstPackages.sh	2014-07-19 17:59:17.334180600 -0400
@@ -21,11 +21,11 @@
 make
 cd ..
 
-# You may not want a writable directory here! In this case substitute
-# the directories 'datagens' and 'dataword' by links to somewhere else.
-cd atlasrep
-chmod 1777 datagens dataword
-cd ..
+# # You may not want a writable directory here! In this case substitute
+# # the directories 'datagens' and 'dataword' by links to somewhere else.
+# cd atlasrep
+# chmod 1777 datagens dataword
+# cd ..
 
 cd Browse
 ./configure
@@ -37,25 +37,25 @@
 # this out until a user complains.
 # It is not possible to move around compiled binaries because these have the
 # path to some data files burned in.
-cd carat
-tar xzpf carat-2.1b1.tgz
-rm -f bin
-ln -s carat-2.1b1/bin bin
-cd carat-2.1b1/functions
-# Install the include Gmp first.
-# (If you have already Gmp on your system, you can delete the file
-# gmp-*.tar.gz and delete the target 'Gmp' from the target 'ALL' in
-# carat-2.1b1/Makefile.)
-tar xzpf gmp-*.tar.gz
-cd ..
-make TOPDIR=`pwd` Links
-# Note that Gmp may use processor specific code, so this step may not be ok
-# for a network installation if you want to use the package on older computers
-# as well.
-make TOPDIR=`pwd` Gmp
-# And now the actual Carat programs.
-make TOPDIR=`pwd` CFLAGS=`-O2`
-cd ../..
+# cd carat
+# tar xzpf carat-2.1b1.tgz
+# rm -f bin
+# ln -s carat-2.1b1/bin bin
+# cd carat-2.1b1/functions
+# # Install the include Gmp first.
+# # (If you have already Gmp on your system, you can delete the file
+# # gmp-*.tar.gz and delete the target 'Gmp' from the target 'ALL' in
+# # carat-2.1b1/Makefile.)
+# tar xzpf gmp-*.tar.gz
+# cd ..
+# make TOPDIR=`pwd` Links
+# # Note that Gmp may use processor specific code, so this step may not be ok
+# # for a network installation if you want to use the package on older computers
+# # as well.
+# make TOPDIR=`pwd` Gmp
+# # And now the actual Carat programs.
+# make TOPDIR=`pwd` CFLAGS=`-O2`
+# cd ../..
 
 cd cohomolo
 ./configure 
@@ -65,7 +65,7 @@
 make 
 cd ..
 
-cd cvec
+cd cvec-*
 ./configure 
 make
 cd ..
@@ -80,10 +80,11 @@
 make
 cd ..
 
-cd float-*
-./configure
-make
-cd ..
+# It's too much trouble to figure out how to build float.
+# cd float-*
+# ./configure
+# make
+# cd ..
 
 cd fplsa
 ./configure ../..
@@ -106,7 +106,7 @@
 make install
 cd ..
 
-cd io
+cd io-*
 ./configure
 make
 cd ..
@@ -117,14 +117,14 @@
 make COPTS="-O2 -g"
 cd ..
 
-#  Easy, if prerequisites are installed. You may get GNU GMP
-#  (http://gmplib.org/) and BLAS (http://www.netlib.org/blas/) 
-#  via packages in your Linux distribution. But you probably need to
-#  install LinBox (http://www.linalg.org/download.html) yourself.
-cd linboxing
-./configure 
-make
-cd ..
+# #  Easy, if prerequisites are installed. You may get GNU GMP
+# #  (http://gmplib.org/) and BLAS (http://www.netlib.org/blas/) 
+# #  via packages in your Linux distribution. But you probably need to
+# #  install LinBox (http://www.linalg.org/download.html) yourself.
+# cd linboxing
+# ./configure 
+# make
+# cd ..
 
 # nq needs GMP. It may be able to find GMP supplied with GAP 
 # automatically. Otherwise, adjust the argument to 'make' accordingly.
@@ -135,7 +135,7 @@
 make
 cd ..
 
-cd orb
+cd orb-*
 ./configure 
 make
 cd ..
@@ -140,20 +141,20 @@
 make
 cd ..
 
-#  see the pargap documentation how to use this 
-#  (in particular, user needs procgroup file)
-#  [more complicated to include EDIM kernel module ...]
-cd pargap
-./configure ../..
-make 
-cp bin/pargap.sh ../../bin/
-cd ..
-rm -f ALLPKG
+# #  see the pargap documentation how to use this 
+# #  (in particular, user needs procgroup file)
+# #  [more complicated to include EDIM kernel module ...]
+# cd pargap
+# ./configure ../..
+# make 
+# cp bin/pargap.sh ../../bin/
+# cd ..
+# rm -f ALLPKG
 
-cd PolymakeInterface
-./configure ../..
-make
-cd ..
+# cd PolymakeInterface
+# ./configure ../..
+# make
+# cd ..
 
 # For XGap the following shared libraries of the X window system must be 
 # installed on your machine together with the development files (header 
