--- origsrc/gap4r6/InstPackages.sh	2013-08-22 18:19:14.893455300 -0400
+++ src/gap4r6/InstPackages.sh	2013-08-23 09:21:44.648644100 -0400
@@ -11,10 +11,10 @@
 # Even if it doesn't work completely automatically for you, you may get 
 # an idea what to do for a complete installation of GAP.
 
-cd ace
-./configure ../..
-make
-cd ..
+# cd ace
+# ./configure ../..
+# make
+# cd ..
 
 # on Linux with gmp in standard place and gap.sh script in standard 
 # path as 'gap' just do 'make', otherwise check installation
