--- origsrc/gap4r6/cnf/configure.in	2013-07-20 16:02:25.000000000 -0400
+++ src/gap4r6/cnf/configure.in	2013-08-18 13:49:24.356476800 -0400
@@ -326,7 +326,7 @@ AC_SUBST(ITANIUMOBJ)
 
 case "$host_os" in
   *cygwin*)
-    AC_DEFINE(SYS_IS_CYGWIN32, 1, [define if this is the Cygwin32 port])
+    AC_DEFINE(SYS_IS_CYGWIN, 1, [define if this is the Cygwin port])
     ;;
   *darwin*)
     AC_DEFINE(SYS_IS_DARWIN, 1, [define if this is the Darwin port])
--- origsrc/gap4r6/src/sysfiles.c	2013-07-20 16:11:08.000000000 -0400
+++ src/gap4r6/src/sysfiles.c	2013-08-18 13:51:20.046280000 -0400
@@ -498,7 +498,7 @@ Obj FuncCrcString( Obj self, Obj str ) {
 #if defined(SYS_IS_DARWIN) && SYS_IS_DARWIN
 # define SYS_INIT_DYNAMIC       "Init__Dynamic"
 #else
-#if defined(SYS_IS_CYGWIN32) && SYS_IS_CYGWIN32
+#if defined(SYS_IS_CYGWIN) && SYS_IS_CYGWIN
 # define SYS_INIT_DYNAMIC       "Init__Dynamic"
 #else
 # define SYS_INIT_DYNAMIC       "_Init__Dynamic"
@@ -699,7 +699,7 @@ Char * SyWinCmd (
     const Char *        bb;             /* pointer into the temporary      */
     Char *              b;              /* pointer into the temporary      */
     UInt                i;              /* loop variable                   */
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     UInt                len1;           /* temporary storage for len       */
 #endif
 
@@ -743,7 +743,7 @@ Char * SyWinCmd (
     /* read the arguments of the answer                                    */
     b = WinCmdBuffer;
     i = len;
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     len1 = len;
     while ( 0 < i ) {
         len = read( 0, b, i );
@@ -897,7 +897,7 @@ Int SyFopen (
         SyExit(2);
       }
 
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     if(strlen(mode) >= 2 && mode[1] == 'b')
        flags |= O_BINARY;
 #endif
