--- origsrc/gap4r7/cnf/configure.in	2014-05-24 16:03:14.000000000 -0400
+++ src/gap4r7/cnf/configure.in	2014-07-14 12:10:55.035668100 -0400
@@ -341,7 +341,7 @@ AC_SUBST(ITANIUMOBJ)
 
 case "$host_os" in
   *cygwin*)
-    AC_DEFINE(SYS_IS_CYGWIN32, 1, [define if this is the Cygwin32 port])
+    AC_DEFINE(SYS_IS_CYGWIN, 1, [define if this is the Cygwin port])
     ;;
   *darwin*)
     AC_DEFINE(SYS_IS_DARWIN, 1, [define if this is the Darwin port])
--- origsrc/gap4r7/pkg/cvec-2.5.1/configure.ac	2014-05-14 09:14:41.000000000 -0400
+++ src/gap4r7/pkg/cvec-2.5.1/configure.ac	2014-07-14 12:19:35.046411100 -0400
@@ -24,9 +24,9 @@ AC_CHECK_SIZEOF([void **])
 AC_CYGWIN
 AM_CONDITIONAL([SYS_IS_CYGWIN], [test "$CYGWIN" = "yes"])
 if test "$CYGWIN" = "yes"; then
-  AC_DEFINE(SYS_IS_CYGWIN32, 1, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 1, are we on CYGWIN?)
 else
-  AC_DEFINE(SYS_IS_CYGWIN32, 0, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 0, are we on CYGWIN?)
 fi
      
 AC_CONFIG_FILES([Makefile])
--- origsrc/gap4r7/pkg/float-0.5.18/m4/ac_sys_is.m4	2014-05-14 09:14:44.000000000 -0400
+++ src/gap4r7/pkg/float-0.5.18/m4/ac_sys_is.m4	2014-07-14 12:23:50.418017500 -0400
@@ -3,9 +3,9 @@ AC_DEFUN([AC_SYS_IS_CYGWIN],[
 AC_CYGWIN
 AM_CONDITIONAL([SYS_IS_CYGWIN], [test "$CYGWIN" = "yes"])
 if test "$CYGWIN" = "yes"; then
-  AC_DEFINE(SYS_IS_CYGWIN32, 1, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 1, are we on CYGWIN?)
 else
-  AC_DEFINE(SYS_IS_CYGWIN32, 0, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 0, are we on CYGWIN?)
 fi
 ])
 
--- origsrc/gap4r7/pkg/fr-2.1.1/cnf/m4/ac_sys_is.m4	2014-05-14 09:14:46.000000000 -0400
+++ src/gap4r7/pkg/fr-2.1.1/cnf/m4/ac_sys_is.m4	2014-07-14 12:24:43.257039700 -0400
@@ -3,9 +3,9 @@ AC_DEFUN([AC_SYS_IS_CYGWIN],[
 AC_CYGWIN
 AM_CONDITIONAL([SYS_IS_CYGWIN], [test "$CYGWIN" = "yes"])
 if test "$CYGWIN" = "yes"; then
-  AC_DEFINE(SYS_IS_CYGWIN32, 1, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 1, are we on CYGWIN?)
 else
-  AC_DEFINE(SYS_IS_CYGWIN32, 0, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 0, are we on CYGWIN?)
 fi
 ])
 
--- origsrc/gap4r7/pkg/io-4.3.1/configure.ac	2014-05-14 09:14:56.000000000 -0400
+++ src/gap4r7/pkg/io-4.3.1/configure.ac	2014-07-14 12:26:14.520259700 -0400
@@ -51,9 +51,9 @@ AC_CHECK_FUNCS([dup2 gethostbyname lchow
 AC_CYGWIN
 AM_CONDITIONAL([SYS_IS_CYGWIN], [test "$CYGWIN" = "yes"])
 if test "$CYGWIN" = "yes"; then
-  AC_DEFINE(SYS_IS_CYGWIN32, 1, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 1, are we on CYGWIN?)
 else
-  AC_DEFINE(SYS_IS_CYGWIN32, 0, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 0, are we on CYGWIN?)
 fi
      
 AC_CONFIG_FILES([Makefile])
--- origsrc/gap4r7/pkg/orb-4.7.1/configure.ac	2014-05-14 09:15:17.000000000 -0400
+++ src/gap4r7/pkg/orb-4.7.1/configure.ac	2014-07-14 12:27:03.273048200 -0400
@@ -24,9 +24,9 @@ AC_CHECK_SIZEOF([void **])
 AC_CYGWIN
 AM_CONDITIONAL([SYS_IS_CYGWIN], [test "$CYGWIN" = "yes"])
 if test "$CYGWIN" = "yes"; then
-  AC_DEFINE(SYS_IS_CYGWIN32, 1, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 1, are we on CYGWIN?)
 else
-  AC_DEFINE(SYS_IS_CYGWIN32, 0, are we on CYGWIN?)
+  AC_DEFINE(SYS_IS_CYGWIN, 0, are we on CYGWIN?)
 fi
      
 AC_CONFIG_FILES([Makefile])
--- origsrc/gap4r7/src/sysfiles.c	2014-07-14 11:54:49.430438700 -0400
+++ src/gap4r7/src/sysfiles.c	2014-07-14 12:10:55.040668400 -0400
@@ -81,7 +81,7 @@ typedef void       sig_handler_t ( int )
 extern int errno;
 #endif
 
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
 # include       <process.h>
 #endif
 
@@ -498,7 +498,7 @@ Obj FuncCrcString( Obj self, Obj str ) {
 #if defined(SYS_IS_DARWIN) && SYS_IS_DARWIN
 # define SYS_INIT_DYNAMIC       "Init__Dynamic"
 #else
-#if defined(SYS_IS_CYGWIN32) && SYS_IS_CYGWIN32
+#if defined(SYS_IS_CYGWIN) && SYS_IS_CYGWIN
 # define SYS_INIT_DYNAMIC       "Init__Dynamic"
 #else
 # define SYS_INIT_DYNAMIC       "_Init__Dynamic"
@@ -699,7 +699,7 @@ Char * SyWinCmd (
     const Char *        bb;             /* pointer into the temporary      */
     Char *              b;              /* pointer into the temporary      */
     UInt                i;              /* loop variable                   */
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     UInt                len1;           /* temporary storage for len       */
 #endif
 
@@ -743,7 +743,7 @@ Char * SyWinCmd (
     /* read the arguments of the answer                                    */
     b = WinCmdBuffer;
     i = len;
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     len1 = len;
     while ( 0 < i ) {
         len = read( 0, b, i );
@@ -897,7 +897,7 @@ Int SyFopen (
         SyExit(2);
       }
 
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
     if(strlen(mode) >= 2 && mode[1] == 'b')
        flags |= O_BINARY;
 #endif
@@ -3110,7 +3110,7 @@ extern char ** environ;
 
 void NullSignalHandler(int scratch) {}
 
-#if SYS_IS_CYGWIN32
+#if SYS_IS_CYGWIN
 
 UInt SyExecuteProcess (
     Char *                  dir,
